PAD L1 updates

woodbe opened the pull request:
Based on the work for the PAD-L2, this is the update for the PAD-L1 changes for the fingerprint

woodbe committed:
Merge pull request #75 from biometricITC/Working

Publication for v2 (matched with v1.1 PPM)

woodbe committed:
Merge pull request #80 from biometricITC/Working

Lab review update

woodbe committed:
Merge pull request #85 from biometricITC/Working

Minor updates on links and tools

woodbe committed:
Merge pull request #87 from biometricITC/Working

TD BIO0004

woodbe committed:
PAD-L1 proposal

Based on the work for the PAD-L2, this is the update for the PAD-L1 changes for the fingerprint

woodbe commented:
This needs work on how to take a latent print I think, and we probably need some additional materials based on that. Any new materials should be added at M.5 or above and then we adjust the L2 materials based on that.

woodbe commented:
This will close #90, close #89, close #88

woodbe committed:
Update to sync everything

This update moved glue into L1, updated the naming so all attacks and the inventory components are uniquely tied to the L1 test set.

The overview has also been updated.

woodbe commented:
@gfiumara @gregott Please check line 122 in the Overview about whether or not the image should be flipped when using a camera. I know it should when using the scanner, but I think if I'm using the camera to image the latent print, I don't want to flip it because I'm taking it flipped already by viewing it from the opposite side.

woodbe committed:
heading update

this will let the L2 overview match better (some sections can be removed more easily so the L2 can focus on the differences while maintaining the same heading structure)

gregott commented:
> @gfiumara @gregott Please check line 122 in the Overview about whether or not the image should be flipped when using a camera. I know it should when using the scanner, but I think if I'm using the camera to image the latent print, I don't want to flip it because I'm taking it flipped already by viewing it from the opposite side.

I believe fingerprint scanners provide the same perspective as you get from an inked fingerprint on the old-fashioned cards.  In other words, the image from a scanner will look just like the image from a latent of the same finger (excepting noise, of course).  If this is true, the scanner image will need the same processing steps as a photo of a latent.

The only exception would be if you are collecting latent prints from a dark surface and are using white dusting powder.  The assumption with scanner images is the ridges are gray and the valleys are white.  When binarizing the scanner image the ridges become black and the valleys stay white.  For latents on dark surfaces with white dusting powder, the ridges will be white (light gray) and the valleys black (dark gray) before the binarization step.  The evaluator will want to take care to end up with black ridges and white valleys after binarization in both cases.

woodbe committed:
typo

woodbe committed:
Update L1-FINGERPRINT_Toolbox_Overview.adoc

Undo one change

woodbe commented:
Thanks, I have undone that particular change.

woodbe committed:
added glue storage

realized this was missing.

woodbe committed:
Update L1-FINGERPRINT_Toolbox_Overview.adoc

woodbe committed:
updates from call

these are to match changes proposed for the L2 testing.

woodbe commented:
> The only exception would be if you are collecting latent prints from a dark surface and are using white dusting powder. The assumption with scanner images is the ridges are gray and the valleys are white. When binarizing the scanner image the ridges become black and the valleys stay white. For latents on dark surfaces with white dusting powder, the ridges will be white (light gray) and the valleys black (dark gray) before the binarization step. The evaluator will want to take care to end up with black ridges and white valleys after binarization in both cases.

So the instructions I have created so far assume black powder since we have control over what they are created on. I guess we could provide white powder as an option, but I think we can probably just minimize the expectations to having a light colored background so we don't have to adjust from what we are already expecting.

gfiumara commented on L1-FINGERPRINT_Toolbox_Overview.adoc:
```diff
@@ -15,131 +15,133 @@ A cast is the PAI, the individual artefact that will be presented to the sensor
 
 Commercial fingerprint scanners are a good source of flat friction ridge images. Images from commercial scanners are generally high quality, distortion free, have good detail, can be of high resolution, have good grayscale contrast and are readily available. It is recommended that a commercial scanner be used for image acquisition. The highest resolution images available should be used for PAD te ...
 
+The commercial fingerprint scanner should be used to check the quality of the latent prints that are acquired, but not as the source of prints for use in the actual tests.
```
```suggestion
The commercial fingerprint scanner capture of a fingerprint should be used to check the quality of the latent prints that are acquired (by ensuring a positive comparison determination is produced by a COTS fingerprint comparison algorithm), but not as the source of prints for use in the actual tests.
```

gfiumara commented on L1-FINGERPRINT_Toolbox_Overview.adoc:
```diff
@@ -15,131 +15,133 @@ A cast is the PAI, the individual artefact that will be presented to the sensor
+To acquire the fingerprint image, the following steps are followed:
 
-As stated above, direct impression materials make excellent molds. They come in varying set times all of which are on the order of a few minutes.
+. The subject's finger is pressed onto the non-porous surface using the similar pressure to the system being tested.
```
```suggestion
. The subject's finger is pressed onto the non-porous surface using similar pressure to the system being tested.
```

gfiumara commented on L1-FINGERPRINT_Toolbox_Overview.adoc:
```diff
@@ -15,131 +15,133 @@ A cast is the PAI, the individual artefact that will be presented to the sensor
-As stated above, direct impression materials make excellent molds. They come in varying set times all of which are on the order of a few minutes.
+. The subject's finger is pressed onto the non-porous surface using the similar pressure to the system being tested.
+. Carefully dip the brush into the container of powder until the brush is well loaded for the type of powder/brush being used.
+. Test the application away from the print until the right amount of powder is applied (not a heavy deposit).
```
```suggestion
. Tap and swirl the brush away from the print to remove excess powder from the brush and to separate the bristles.
```

gfiumara commented on L1-FINGERPRINT_Toolbox_Overview.adoc:
```diff
@@ -15,131 +15,133 @@ A cast is the PAI, the individual artefact that will be presented to the sensor
-Fingerprint image acquisition for flat PAIs is the same for all attacks. An image of the subject’s finger is captured using a high quality commercial fingerprint scanner. The image can be from a slap or rolled image. If a rolled image is acquired, the evaluator may want to crop the fingerprint image to make a smaller mold containing only the central part of the fingerprint that would usually be i ...
-
-The image then needs to be processed such that the resultant image contains only black (0 in the 8-bit grayscale color space) and white (255 in the 8-bit grayscale color space) pixels. This can be done by hand using photo editing software suitable for pixel level adjustments or in an automated fashion using binarization software. The resulting enhanced images should have black ridges and white va ...
+The image (whether from the scanner or captured from the latent print) then needs to be processed such that the resultant image contains only black (0 in the 8-bit grayscale color space) and white (255 in the 8-bit grayscale color space) pixels. This can be done by hand using photo editing software suitable for pixel level adjustments or in an automated fashion using binarization software. The re ...
```
```suggestion
The image (whether from the scanner or captured from the latent print) then needs to be binarized to only black (0 in the 8-bit grayscale color space) and white (255 in the 8-bit grayscale color space) pixels. This can be done by hand using photo editing software suitable for pixel level adjustments or in an automated fashion using binarization software. The resulting enhanced images should have black ridges and white valleys. Pores and creases are also white. The resultant image should be life-sized and remain at the highest resolution available. Resizing the image shall not use interpolation without justification approved by the BIT.
```

gfiumara replied:
May be worded better, but trying to double-check the lift for quality after binarization.

```suggestion
The image (whether from the scanner or captured from the latent print) then needs to be processed such that the resultant image contains only black (0 in the 8-bit grayscale color space) and white (255 in the 8-bit grayscale color space) pixels. This can be done by hand using photo editing software suitable for pixel level adjustments or in an automated fashion using binarization software. The resulting enhanced images should have black ridges and white valleys. Pores and creases are also white. The resultant image should be life-sized and remain at the highest resolution available. Resizing the image shall not use interpolation without justification approved by the BIT.

If excessive powder or background artifacts involved in the acquisition of the fingermark create excessive false ridge detail in the resulting binarized image, the steps for acquiring a latent print shall be repeated.
```

gfiumara commented on L1-FINGERPRINT_Toolbox_Overview.adoc:
```diff
@@ -15,131 +15,133 @@ A cast is the PAI, the individual artefact that will be presented to the sensor
+The other method for capturing the print is to pull the print powder using some sort of adhesive tape and then placing this onto a fingerprint scanner to capture the image. 
 
-Section 5.1.1 Preparation of the Biometric PAD Toolbox Overview should be reviewed for further details about how to ensure high quality samples.
+When supported by the scanner, the image file received shall be in bitmap, TIFF, RAW, or other lossless file formats. Image files using the WSQ format at a bitrate of 2.25 (≅5:1 compression) are acceptable if lossless file formats are not available from the scanner.
```
```suggestion
When supported by the scanner, the image file received shall be in bitmap, TIFF, RAW, or other lossless file formats. Image files using the WSQ format at a bitrate of 2.25 (≅5:1 compression) are acceptable if lossless file formats are not available from the scanner.

==== Quality of the latent print
The resulting image shall be a high enough quality representation of the source friction ridge skin as to be suitable for PAI creation. Upon analysis of the fingermark, the examiner shall observe a single non-complex impressionfootnote:[https://www.aafs.org/asb-standard/best-practice-recommendation-analysis-friction-ridge-impressions]. If a low or high complexity fingermark is recovered, the steps for acquiring a latent print shall be repeated.
```

gfiumara commented on L1-FINGERPRINT_Toolbox_Overview.adoc:
```diff
@@ -15,131 +15,133 @@ A cast is the PAI, the individual artefact that will be presented to the sensor
+=== Lab preparation using a scanner
 
-===== Overview
+The lab is encouraged to use a scanner to be able to compare the captured print with a direct input as a general measure of quality. Information seen from the direct print shall not be transferred to the captured print (for example seeing that a ridge is broken in the latent when compared to the direct scan shall not be used to correct the latent image).
```
```suggestion
The lab is encouraged to use a scanner to be able to compare the captured print with a direct input as a general measure of quality. Information seen from the direct print shall not be transferred to the captured print (for example seeing that a ridge is broken in the latent when compared to the direct scan shall not be used to correct the latent image). The comparison shall be performed prior to inverting the binarized image.
```

gfiumara commented on L1-FINGERPRINT_Toolbox_References.adoc:
```diff
@@ -24,4 +24,16 @@ https://www.nist.gov/services-resources/software/nist-biometric-image-software-n
+
+[9] [Developing Latent Prints] Trozzi, T., Schwartz, R., Hollars, M, 2000. Processing Guide for Developing Latent Prints. FBI.
+
+https://archives.fbi.gov/archives/about-us/lab/forensic-science-communications/fsc/jan2001/lpu.pdf
```
```suggestion
https://archives.fbi.gov/archives/about-us/lab/forensic-science-communications/fsc/jan2001/lpu.pdf

[10] [Analysis BPR] ANSI/ASB Best Practice Recommendation 165: Best Practice Recommendation for Analysis of Friction Ridge Impressions. First Edition.

https://www.aafs.org/asb-standard/best-practice-recommendation-analysis-friction-ridge-impressions
```

woodbe committed:
Update L1-FINGERPRINT_Toolbox_Overview.adoc

Co-authored-by: Greg Fiumara <gregory.fiumara@nist.gov>

woodbe committed:
Update L1-FINGERPRINT_Toolbox_Overview.adoc

Co-authored-by: Greg Fiumara <gregory.fiumara@nist.gov>

woodbe committed:
Update L1-FINGERPRINT_Toolbox_Overview.adoc

Co-authored-by: Greg Fiumara <gregory.fiumara@nist.gov>

woodbe committed:
Update L1-FINGERPRINT_Toolbox_Overview.adoc

Co-authored-by: Greg Fiumara <gregory.fiumara@nist.gov>

woodbe committed:
Update L1-FINGERPRINT_Toolbox_Overview.adoc

Co-authored-by: Greg Fiumara <gregory.fiumara@nist.gov>

woodbe committed:
Update L1-FINGERPRINT_Toolbox_Overview.adoc

Co-authored-by: Greg Fiumara <gregory.fiumara@nist.gov>

woodbe committed:
Update L1-FINGERPRINT_Toolbox_References.adoc

Co-authored-by: Greg Fiumara <gregory.fiumara@nist.gov>

woodbe committed:
Update L1-FINGERPRINT_Toolbox_Overview.adoc

To add from one of the suggestions that wouldn't allow me to commit (I think because it sort of lost track of the location based on earlier commits that were added)

woodbe committed:
Updates from the call

Updates to the issues raised on the call.

n-kai committed:
Update L1-FINGERPRINT_Toolbox_Overview.adoc

n-kai committed:
Update L1-FINGERPRINT_Toolbox_References.adoc

n-kai reviewed (changes requested):
I found some errors and created PR #93 .

woodbe committed:
Merge pull request #93 from biometricITC/Minor-updates-for-PAD-L1-updates

Minor updates for pad L1 updates

woodbe committed:
Update L1-FINGERPRINT_Toolbox_References.adoc

adding fingerprint spoof paper

woodbe commented:
The latest addition is to close #90.

n-kai reviewed (approved):
I am not an expert in this area but I can't find any inconsistency between recipes in toolbox and research papers I suggested to add to the Toolbox_reference.
